Geometry/CaloGeometry/src/CaloGenericDetId.cc

```diff
@@ -64,41 +74,25 @@ CaloGenericDetId::validDetId() const
    }
    else
    {
-      if( isEE() )
-      {
-	 const EEDetId eeid ( rawId() ) ;
-	 returnValue = EEDetId::validDetId( eeid.ix(), 
-					    eeid.iy(),
-					    eeid.zside() ) ;
-      }
-      else
-      {
-	 if( isES() )
-	 {
-	    const ESDetId esid ( rawId() ) ;
-	    returnValue = ESDetId::validDetId( esid.strip(),
-					       esid.six(),
-					       esid.siy(), 
-					       esid.plane(),
-					       esid.zside() ) ;
-	 }
-	 else
-	 {
-	   if (det() == DetId::Hcal) { 
-	     std::cerr << "No support for HB/HE/HO/HF in CaloGenericDetId" << std::endl;
-	     throw cms::Exception("No support");
-
-	     returnValue = false;
+	 if( isEE() )
+	   {
+	     const EEDetId eeid ( rawId() ) ;
+	     returnValue = EEDetId::validDetId( eeid.ix(), 
+						eeid.iy(),
+						eeid.zside() ) ;
 	   }
-	   else
-	     {
-	       if( isZDC() )
```

Comment:
@ianna - it looks like this was lost in the rebase. I'll fix it, thanks for the catch. (I'll also rewrite this part to use "else if" instead of nested if-else sets.)